### PR TITLE
JS solution and tests for Issue #18 - Find missing number in array of…

### PR DESCRIPTION
**Pull request by zachnagatani**
… consecutively increasing numbers

### CODE REVIEW

File: test/18.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+let solution = require('../solutions/18').solution;
+//solution = require('../yourSolution').solution;
+
+describe('Find missing number in array', () => {
+  it('should return 7 for [5,6,8,9]', () => {
+    const result = solution([5,6,8,9]);
+    expect(result).to.equal(7);
+  });
+
+  it('should return 94 for [89,90,91,92,93]', () => {
```
Comment (songz) [inline test/18.js]: description incomplete. [...91, 92, 93, 95]

File: solutions/18.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Finds and returns the missing number from an array of consecutively increasing numbers
+ * @param {Number[]} arr - An array of consecutively increasing numbers
+ * @returns {Number} missingNum - the number missing from the input array
+ */
+const solution = arr => {
+    let missingNum = null,
+        i = 0;
+
+    while (!missingNum && i < arr.length - 1) {
```
Comment (songz) [inline solutions/18.js]: food for thought. Since there is only 1 missing number, and your goal is to return that number, you might not even need the missingNum variable. start `i = 1`.

while (i < arr.length -1 && arr[i-1] +1 === arr[i] )....

You don't have to make this change, just food for thought. Good job!

File: test/18.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    const result = solution([89,90,91,92,93,95]);
+    expect(result).to.equal(94);
+  });
+
+  it('should return null if there is no missing number', () => {
+    const result = solution([1,2,3,4,5]);
+    expect(result).to.equal(null);
+  });
+});
```
Comment (songz) [inline test/18.js]: could you add a test case for negative numbers please? thx.

**Comment by zachnagatani** [comment]
@songz Thanks for the feedback! I liked your approach to the solution, so I just added it as an alternative one in the bottom of the file. Had to make one change to the while condition so that I could also return null if no missing number is found.

File: solutions/18.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+// output: 7
+
+// Zach Nagatani
+
+/**
+ * Finds and returns the missing number from an array of consecutively increasing numbers
+ * @param {Number[]} arr - An array of consecutively increasing numbers
+ * @returns {Number|null} missingNum - the number missing from the input array or null if none is found
+ */
+// const solution = arr => {
```
Comment (songz) [inline solutions/18.js]: instead of commenting out previous solution, could you rename them? `solution`, `solution1`,  and export  all the solutions.

Comment (zachnagatani) [inline solutions/18.js]: Got it!

**Comment by songz** [comment]
CONGRATS! This is a great solution to issue #18
